lib/src/config/deprecated_option.dart

```diff
@@ -0,0 +1,22 @@
+import 'package:meta/meta.dart';
+
+/// Class representing deprecated config option
+@immutable
+class DeprecatedOption {
+  /// Version number until which we support obsolete option
+  final String supportUntilVersion;
+
+  /// Obsolete option name
+  final String deprecated;
+
+  /// Modern option name (optional)
+  final String? modern;
+
+  const DeprecatedOption({
+    required this.supportUntilVersion,
+    required this.deprecated,
+    this.modern,
+  });
+}
+
+const Iterable<DeprecatedOption> deprecatedOptions = [];
```

Comment:
Is it intensional to have an empty list?

Reply:
sure, for easy maintenance of deprecated options, we will just need to add them to this array